src/nvim/normal.c

```diff
@@ -2939,6 +2939,9 @@ void check_visual_highlight(void)
  */
 void end_visual_mode(void)
 {
+  if (exiting) {
+    return;
+  }
```

Comment:
Was this from a vim patch? This seems too localized. How did we reach this? From `normal_handle_special_visual_command()` ? 

Until now the only places we checking `exiting` are (1) to avoid scheduling new events, and (2) to avoid showing user messages.